marlowe-dashboard-client/src/Capability/Wallet.purs

```diff
@@ -0,0 +1,82 @@
+module Capability.Wallet
+  ( class MonadWallet
+  , createWallet
+  , submitWalletTransaction
+  , getWalletPubKey
+  , updateWalletPaymentWithChange
+  , getWalletSlot
+  , getWalletTransactions
+  , getWalletTotalFunds
+  , signTransaction
+  ) where
+
+import Prelude
+import AppM (AppM)
+import Bridge (toBack, toFront)
+import Capability.Ajax (runAjax)
+import Control.Monad.Except (lift)
+import Data.Json.JsonTuple (JsonTuple)
+import Data.Map (Map)
+import Data.Newtype (unwrap)
+import Halogen (HalogenM)
+import MainFrame.Types (WebData)
+import Marlowe.Semantics (Assets, PubKey, Slot)
+import Plutus.PAB.Webserver (getWalletByWalletIdOwnoutputs, getWalletByWalletIdOwnpublickey, getWalletByWalletIdTotalfunds, getWalletByWalletIdWalletslot, postWalletByWalletIdSign, postWalletByWalletIdSubmittxn, postWalletByWalletIdUpdatepaymentwithchange, postWalletCreate)
+import Plutus.V1.Ledger.Tx (Tx, TxOutRef, TxOutTx)
+import Wallet.Types (Payment)
+import WalletData.Types (Wallet)
+
+-- The PAB PSGenerator (using Servant.PureScript) automatically generates a PureScript module with
+-- functions for calling all Wallet API endpoints. This `MonadWallet` class wraps these up in a
+-- 'capability' monad (https://thomashoneyman.com/guides/real-world-halogen/push-effects-to-the-edges/)
+-- with some nicer names and type signatures, mapping the result to WebData.
+class
+  Monad m <= MonadWallet m where
+  createWallet :: m (WebData Wallet)
+  submitWalletTransaction :: Tx -> Wallet -> m (WebData Unit)
+  getWalletPubKey :: Wallet -> m (WebData PubKey)
+  updateWalletPaymentWithChange :: JsonTuple Assets Payment -> Wallet -> m (WebData Payment)
+  getWalletSlot :: Wallet -> m (WebData Slot)
+  getWalletTransactions :: Wallet -> m (WebData (Map TxOutRef TxOutTx))
+  getWalletTotalFunds :: Wallet -> m (WebData Assets)
+  signTransaction :: Tx -> Wallet -> m (WebData Tx)
```

Comment:
Not sure, but `signTransaction` seems low level and could be internal for other endpoints to call. For example if `submitWalletTransaction` needs the transaction to be signed, let's do that there instead of exposing this and relying on the user of this API to remember to sign.